tools/boards.txt.py

```diff
@@ -1153,24 +1155,31 @@ def flash_size (size_bytes, display, optname, ld, desc, max_upload_size, spiffs_
-    f1m.update( flash_size(0x100000,   '1M', '1M256',   'eagle.flash.1m256.ld',   '256K SPIFFS', 761840,   0xBB000,   0x40000, 4096))
-    f1m.update( flash_size(0x100000,   '1M', '1M512',   'eagle.flash.1m512.ld',   '512K SPIFFS', 499696,   0x7B000,   0x80000, 8192))
-    f2m =       flash_size(0x200000,   '2M', '2M',      'eagle.flash.2m.ld',        '1M SPIFFS', 1044464, 0x100000,   0xFB000, 8192)
-    f4m =       flash_size(0x400000,   '4M', '4M1M',    'eagle.flash.4m1m.ld',      '1M SPIFFS', 1044464, 0x300000,   0xFB000, 8192)
-    f4m.update( flash_size(0x400000,   '4M', '4M2M',    'eagle.flash.4m2m.ld',      '2M SPIFFS', 1044464, 0x200000,  0x1FB000, 8192))
-    f4m.update( flash_size(0x400000,   '4M', '4M3M',    'eagle.flash.4m.ld',        '3M SPIFFS', 1044464, 0x100000,  0x2FB000, 8192))
-    f8m =       flash_size(0x800000,   '8M', '8M7M',    'eagle.flash.8m.ld',        '7M SPIFFS', 1044464, 0x100000,  0x6FB000, 8192)
-    f16m =      flash_size(0x1000000, '16M', '16M15M',  'eagle.flash.16m.ld',      '15M SPIFFS', 1044464, 0x100000,  0xEFB000, 8192)
+    # notes:
+    #     spiffs:start = max-upload + 4112
+    #     spiffs:start is always provided for reference and not used when size=0
+    #
+    #                      size      showed / ld name         descr.  max-upload spiffs:start size blocksize
+    #
+    f512 =      flash_size(0x80000,  '512K', '512k0',     'no SPIFFS',  499696,  0x7B000)
+    f512.update(flash_size(0x80000,  '512K', '512k32',   '32K SPIFFS',  466928,  0x73000,    0x8000, 4096))
+    f512.update(flash_size(0x80000,  '512K', '512k64',   '64K SPIFFS',  434160,  0x6B000,   0x10000, 4096))
+    f512.update(flash_size(0x80000,  '512K', '512k128', '128K SPIFFS',  368624,  0x5B000,   0x20000, 4096))
+    f1m =       flash_size(0x100000,   '1M', '1m0',       'no SPIFFS', 1023984,  0xFB000)
+    f1m.update( flash_size(0x100000,   '1M', '1m64',     '64K SPIFFS',  958448,  0xEB000,   0x10000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m128',   '128K SPIFFS',  892912,  0xDB000,   0x20000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m144',   '144K SPIFFS',  876528,  0xD7000,   0x24000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m160',   '160K SPIFFS',  860144,  0xD3000,   0x28000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m192',   '192K SPIFFS',  827376,  0xCB000,   0x30000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m256',   '256K SPIFFS',  761840,  0xBB000,   0x40000, 4096))
+    f1m.update( flash_size(0x100000,   '1M', '1m512',   '512K SPIFFS',  499696,  0x7B000,   0x80000, 8192))
+    f2m =       flash_size(0x200000,   '2M', '2m0',      'no SPIFFS',  1044464, 0x100000)
+    f2m.update( flash_size(0x200000,   '2M', '2m',       '1M SPIFFS',  1044464, 0x100000,   0xFB000, 8192))
+    f4m =       flash_size(0x400000,   '4M', '4m1m',     '1M SPIFFS',  1044464, 0x300000,   0xFB000, 8192)
```

Comment:
I suggest a 4M0, at least for testing purposes if nothing else. I don't know if it's the case for everyone else, but I have a whole lot of boards and they're all 4MB flash.

Reply:
added in #5103